grafana/src/main/java/org/apache/hertzbeat/grafana/service/DatasourceService.java

```diff
@@ -0,0 +1,140 @@
+    private String victoriaMetricsUrl;
+
+    private final GrafanaConfiguration grafanaConfiguration;
+    private final VictoriaMetricsProperties warehouseProperties;
+    private final RestTemplate restTemplate;
+
+    @Autowired
+    public DatasourceService(
+            GrafanaConfiguration grafanaConfiguration,
+            VictoriaMetricsProperties warehouseProperties,
+            RestTemplate restTemplate
+    ) {
+        this.grafanaConfiguration = grafanaConfiguration;
+        this.warehouseProperties = warehouseProperties;
+        this.restTemplate = restTemplate;
+    }
+
+    @PostConstruct
+    public void init() {
+        this.grafanaUrl = grafanaConfiguration.getUrl();
+        this.username = grafanaConfiguration.getUsername();
+        this.password = grafanaConfiguration.getPassword();
+        this.prefix = grafanaConfiguration.getPrefix();
+        this.victoriaMetricsUrl = warehouseProperties.url();
+    }
+
+    /**
+     * Creates a new datasource in Grafana.
+     *
+     * @return ResponseEntity containing the response from Grafana
+     */
+    public ResponseEntity<String> createDatasource() {
+        String url = String.format(prefix + CREATE_DATASOURCE_API, username, password, grafanaUrl);
+
+        HttpHeaders headers = createHeaders();
+
+        String body = String.format(
+                "{\"name\":\"%s\",\"type\":\"%s\",\"access\":\"%s\",\"url\":\"%s\",\"basicAuth\":%s}",
+                DATASOURCE_NAME, DATASOURCE_TYPE, DATASOURCE_ACCESS, victoriaMetricsUrl, false
+        );
+
+        HttpEntity<String> entity = new HttpEntity<>(body, headers);
+
+        try {
+            ResponseEntity<String> response = restTemplate.postForEntity(url, entity, String.class);
+            if (response.getStatusCode().is2xxSuccessful()) {
+                log.info("Create datasource success");
+            }
+            return response;
+        } catch (Exception ex) {
+            log.error("Create datasource error", ex);
+            throw new RuntimeException("Create datasource error", ex);
+        }
+    }
+
+    /**
+     * Deletes a datasource in Grafana.
+     *
+     * @return ResponseEntity containing the response from Grafana
+     */
+    public ResponseEntity<String> deleteDatasource() {
+        String url = String.format(prefix + DELETE_DATASOURCE_API, username, password, grafanaUrl, DATASOURCE_NAME);
+
+        HttpHeaders headers = createHeaders();
+
+        HttpEntity<Void> entity = new HttpEntity<>(headers);
+
+        try {
+            ResponseEntity<String> response = restTemplate.exchange(url, HttpMethod.DELETE, entity, String.class);
+            if (response.getStatusCode().is2xxSuccessful()) {
+                log.info("Delete datasource success");
+            }
+            return response;
+        } catch (Exception ex) {
+            log.error("Delete datasource error", ex);
+            throw new RuntimeException("Delete datasource error", ex);
+        }
+    }
+
```

Comment:
hi, if the datasource is manually deleted by the user in Grafana, the status code through Grafana's HTTP API will be 404, an exception will be thrown, and the manager will not be able to be started normally next time

Reply:
ok, I will fix it.